### PR TITLE
Add day selection screen - part 1

### LINKED ISSUE #14705: Android: Blogging Reminders: select days screen
Add a screen where bloggers can set/edit blogging reminders (see screenshot below: it's iOS based for now).

<p align=center>

</p>

### PR DESCRIPTION
**Pull request by planarvoid**
Fixes #14705 

This PR adds functionality to select days for blogging reminders. This is only the first part of the UI changes because the code changes started to grow too much. The functionality to select a day for a reminder is there but it doesn't trigger a notification yet. 

What's missing
- polish the UI - especially fix the Diff utils for the day buttons so they don't all get refreshed when one is selected and the margins. Let's leave the design review for the second part.
- add the missing views under the Day buttons block

To test:
- Make sure the Blogging reminders flag is turned on
- Go to My Site/Settings
- Click on Blogging goals
- Notice the new updated UX
- Click on a few days 
- Confirm selection
- Notice the summary of the settings item is updated
- Click on a few Blogging goals
- Close bottom sheet
- Notice the changes are discarded and the summary is not updated

To test: 
- Turn on the app with "US English" language
- Notice the Days start with Sunday
- Change locale to some other language that starts week with Monday (for example Czech)
- Notice the Days start with Monday
- Check the same days as before are selected - only the days are shifted

Screenshot 2021-06-02 at 18 11 13

## Regression Notes
1. Potential unintended areas of impact
None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/DaysProvider.kt
Comment (renanferrari) [inline WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/DaysProvider.kt:25]: Even though this works pretty well (as the logic inside `DateProvider` does too), we should probably avoid using `Calendar` around this feature once #14698 gets merged and use `DayOfWeek` instead, since that's already being used in some places.

I know those parts of the code don't interact with each other right now, but I'm just trying to be on the careful side, since those two APIs have some significant differences when it comes to days of week (`Calendar.SUNDAY == 1` and `DayOfWeek.SUNDAY == 7` for example).


If we decide to start using desugaring within FluxC as well, then I guess this would actually be a requirement.

For now, I'd just add a `// TODO` on the top of this block to remind us of doing that. Let me know what you think!

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/DaysProvider.kt
Comment (renanferrari) [inline WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/DaysProvider.kt:33]: Also, if we start using `DayOfWeek` within our `BloggingRemindersModel` I think this logic could be simplified quite a bit (maybe even removed entirely).

**Review by renanferrari — COMMENTED** [review]
Thanks for the changes! 

The app is working as expected for the most part, except that I see these weird animations when opening/closing the bottom sheet:

https://user-images.githubusercontent.com/830056/120874909-307dc780-c57f-11eb-9fa8-4830964843cd.mp4

Are you planning to tackle that as part of the UI polish in the second part?

As for the code, it is looking good overall, I just have a few observations which I've shared below.

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/DaysProvider.kt
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/DaysProvider.kt:25]: that's a good point. I agree 👍 . We can probably add a TODO and create a task to rewrite this once that PR is merged 👍

**Comment by planarvoid** [comment]
Thanks for the review @renanferrari . I've added the TODO and fixed a lint issue. I was wondering about that UI behaviour. I couldn't reproduce it either my emulator nor my pixel. Do you have repro steps? Or does it always just happen to you?

**Review by renanferrari — APPROVED** [review]
Thanks for the changes!

>I was wondering about that UI behaviour. I couldn't reproduce it either my emulator nor my pixel.

I actually tried this again today and couldn't reproduce it myself (using the same device and everything), so I don't know what that was about 🤷‍♂️

I'm gonna merge this and if we come across that bug again we can open a separate issue for it.

**Pull request merged**
